File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/data/items/effects/ObjectEffectCreature.scala





















// Generated on 05/03/2013 10:35:06
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait ObjectEffectCreature extends ObjectEffect {
  override val typeId = 71

  protected trait Type extends ObjectEffect.Type with Serializable {
    override val typeId = ObjectEffectCreature.typeId

    val monsterFamilyId: Short

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeShort(monsterFamilyId)
    }

  }

  protected def create(actionId: Short, monsterFamilyId: Short): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val actionId = buf.readShort()
    val monsterFamilyId = buf.readShort()

    create(actionId, monsterFamilyId)
  }

}

object ObjectEffectCreature extends ObjectEffectCreature {
  case class TypeImpl(actionId: Short, monsterFamilyId: Short) extends Type
  override def create(actionId: Short, monsterFamilyId: Short) = TypeImpl(actionId, monsterFamilyId)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(actionId: Short, monsterFamilyId: Short) = create(actionId, monsterFamilyId)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/ObjectEffectCreature.java


// Generated on 05/08/2013 19:38:02
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ObjectEffectCreature extends ObjectEffect {
    public static final short TYPE_ID = 71;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public short monsterFamilyId;
    
    public ObjectEffectCreature() { }
    
    public ObjectEffectCreature(short actionId, short monsterFamilyId) {
        super(actionId);
        this.monsterFamilyId = monsterFamilyId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeShort(monsterFamilyId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        monsterFamilyId = buf.readShort();
        if (monsterFamilyId < 0)
            throw new RuntimeException("Forbidden value on monsterFamilyId = " + monsterFamilyId + ", it doesn't respect the following condition : monsterFamilyId < 0");
    }
    
}
